custom_endpoint/src/CurrentUserService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\custom_endpoint;
+
+use Drupal\Core\Session\AccountInterface;
+
+class CurrentUserService {
+
+  protected $currentUser;
+
+  /**
+   * Custom endpoint service controller
+   *
+   * @param \Drupal\Core\Session\AccountInterface
+   */
+  public function __construct(AccountInterface $current_user) {
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * Return current user data
+   * or return an error message.
+   */
+  public function getResults(): array {
+    //Try to get user data
+    try {
+      $username = $this->currentUser->getAccountName();
+      $created = $this->currentUser->get['created'];
```

Comment:
instead use:
`$this->currentUser->get('created')->value`
or
`$this->currentUser->get('created')->getValue()`

Reply:
Added ->value

For some reasion getValue() didn't work.